core/services/ocr2/plugins/ocr2keeper/evm21/upkeepstate/store.go

```diff
@@ -103,19 +121,38 @@ func (u *upkeepStateStore) Close() error {
 	return nil
 }
 
+// Name provides the service name
+func (u *upkeepStateStore) Name() string {
+	return u.lggr.Name()
+}
+
+// HealthReport provides a health report on the service
+func (u *upkeepStateStore) HealthReport() map[string]error {
+	return map[string]error{u.Name(): u.StartStopOnce.Healthy()}
+}
+
 // SelectByWorkIDs returns the current state of the upkeep for the provided ids.
 // If an id is not found, the state is returned as StateUnknown.
 // We first check the cache, and if any ids are missing, we fetch them from the scanner.
-// TODO: fetch from DB
 func (u *upkeepStateStore) SelectByWorkIDsInRange(ctx context.Context, start, end int64, workIDs ...string) ([]ocr2keepers.UpkeepState, error) {
 	states, ok := u.selectFromCache(workIDs...)
 	if ok {
 		// all ids were found in the cache
 		return states, nil
 	}
+
+	// fetch values from chain to populate the cache with missing values
 	if err := u.fetchPerformed(ctx, start, end); err != nil {
 		return nil, err
 	}
+
+	// fetch values from the db to populate the cache with missing values here
+	if err := u.fetchFromDB(ctx, workIDs...); err != nil {
```

Comment:
Can be optimized: instead of workIDs, use a filtered list of only those that were not found in cache (the ones that has `unknown` state, you can loop over `states` for that)

either do it or write a TODO note